core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,910 @@
+                    .remove("key2")
+                    .set("key2", "z")
+                    .commit())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot remove and update the same key: key2");
+  }
+
+  @Test
+  public void replaceViewVersion() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    SQLViewRepresentation spark =
+        ImmutableSQLViewRepresentation.builder()
+            .dialect("spark")
+            .sql("select * from ns.tbl")
+            .build();
+
+    SQLViewRepresentation trino =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.tbl")
+            .dialect("trino")
+            .build();
+
+    View view =
+        catalog()
+            .buildView(identifier)
+            .withSchema(SCHEMA)
+            .withDefaultNamespace(identifier.namespace())
+            .withQuery(trino.dialect(), trino.sql())
+            .withQuery(spark.dialect(), spark.sql())
+            .create();
+
+    assertThat(catalog().viewExists(identifier)).as("View should exist").isTrue();
+
+    ViewVersion viewVersion = view.currentVersion();
+    assertThat(viewVersion.representations()).hasSize(2).containsExactly(trino, spark);
+
+    // uses a different schema and view representation
+    view.replaceVersion()
+        .withSchema(OTHER_SCHEMA)
+        .withQuery(trino.dialect(), trino.sql())
+        .withDefaultCatalog("default")
+        .withDefaultNamespace(identifier.namespace())
+        .commit();
+
+    // history and view versions should reflect the changes
+    View updatedView = catalog().loadView(identifier);
+    assertThat(updatedView.history())
+        .hasSize(2)
+        .element(0)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(viewVersion.versionId());
+    assertThat(updatedView.history())
+        .element(1)
+        .extracting(ViewHistoryEntry::versionId)
+        .isEqualTo(updatedView.currentVersion().versionId());
+    assertThat(updatedView.schemas())
+        .hasSize(2)
+        .containsKey(SCHEMA.schemaId())
+        .containsKey(OTHER_SCHEMA.schemaId());
+    assertThat(updatedView.versions())
+        .hasSize(2)
+        .containsExactly(viewVersion, updatedView.currentVersion());
+
+    ViewVersion updatedViewVersion = updatedView.currentVersion();
+    assertThat(updatedViewVersion).isNotNull();
+    assertThat(updatedViewVersion.versionId()).isEqualTo(2);
+    assertThat(updatedViewVersion.summary()).hasSize(1).containsEntry("operation", "replace");
+    assertThat(updatedViewVersion.operation()).isEqualTo("replace");
+    assertThat(updatedViewVersion.representations()).hasSize(1).containsExactly(trino);
+    assertThat(updatedViewVersion.schemaId()).isEqualTo(OTHER_SCHEMA.schemaId());
```

Comment:
Noting that all of the places where this assertion is present should be fixed.

Reply:
done